### DISCUSSION TITLE
Can deckgl set a new property named onViewportChange just like onViewChange?

### DISCUSSION THREAD

**Discussion by Awen-hub** [visgl/deck.gl#7426] [Ideas]
@Pessimistress 

reference issue: [The description](https://github.com/visgl/deck.gl/issues/7420)

Some additional description for my condition:

I just want do use the `viewport` like this
```
const [viewPort, setViewPort] = useState(null)

return(
  <>
  <DeckGL
    onViewportChange={setViewPort}
  />
  <OtherComponent viewPort={viewPort} />
  </>
)

```

There is some reason I can not use `OtherComponent` as DeckGL's child component. So your solution is not the best way for my problem. Unfortunately, it is just the only solution now. In general, I think `onViewportChange` is necessary in this condition. In addition, setting `viewport` as a parameter of `onViewChange` is also a good choice.

**Comment by Pessimistress** [comment]
You never explained why you can't use existing solutions other than that you don't want to. It does not make a convincing argument why there should be a new feature.

Anyways you can construct your own viewport instance from the view state from `onViewStateChange`. https://deck.gl/docs/api-reference/core/web-mercator-viewport

**Reply by Awen-hub** [reply]
The reason is too complex but in short just for performance. Fortunately, `WebMercatorViewport` solved my problem. In this way, I can get a viewport created with viewstate but not from Deckgl. Thanks.
